apollo-router/src/services/connector_service.rs

```diff
@@ -46,10 +46,9 @@ impl tower::Service<ConnectRequest> for ConnectorService {
             .get(&request.service_name)
             .cloned();
 
-        let http_client_factory = connector
-            .as_ref()
-            .map(|c| c.id.subgraph_name.clone())
-            .and_then(|subgraph_name| self.http_service_factory.get(&subgraph_name.to_string()))
+        let http_client_factory = self
+            .http_service_factory
+            .get(&request.service_name.to_string())
```

Comment:
this makes sense because we haven't changed the code that creates the service factory to know about the service_name/connector/subgraph distinction. but currently the HTTP client configuration is by subgraph name, so we'll need to figure out how to map the config to the appropriate HTTP Service